### PR TITLE
Concurrent cursor support multiple formats

### PR DESCRIPTION
**Pull request by maxi297**
## What
Following [the PR to fix SDM](https://github.com/airbytehq/airbyte/pull/48344), we have seen that there was a CDK error popping up related to pendulum not being able to parse a datetime. 

## How
Support the same datetime parsing than `DatetimeBasedCursor` as part of the `CustomFormatConcurrentStreamStateConverter`

## User Impact
We should be able to unblock the release of datascope

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by bnchrch — APPROVED** [review]
LGTM but ill defer to @brianjlai for final approval

**Review by brianjlai — APPROVED** [review]
lgtm!

**Comment by brianjlai** [comment]
small nit, but can we also update https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/cdk-migrations.md to say `Upgrading to 6.0.0`? I forgot to change it from `6.x.x` where i did the original merge

**Comment by maxi297** [comment]
> small nit, but can we also update https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/cdk-migrations.md to say Upgrading to 6.0.0? I forgot to change it from 6.x.x where i did the original merge

I don't want to wait for CI again but I'll put this on my todo list for another PR

**Pull request merged**
